test/iontests.ts

```diff
@@ -45,65 +46,50 @@ function exhaust(reader: Reader) {
     }
 }
 
-function loadValues(path: string, newWriter: () => Writer) {
-    let reader = ion.makeReader(getInput(path));
-    let values: Uint8Array[][] = [];
-    let containerIdx = 0;
-
-    for (let topLevelType; topLevelType = reader.next();) {
-        if (!topLevelType.isContainer) {
-            throw Error('Expected top-level value to be a container, was ' + topLevelType);
-        }
-
-        reader.stepIn();
-        values.push([]);
-
-        for (let type; type = reader.next();) {
-            let writer = newWriter();
-            writer.writeValue(reader);
-            writer.close();
-            values[containerIdx].push(writer.getBytes());
+function equivsTest(path: string, expectedEquivalence = true, equivsTimelines? : boolean) {
+    let bytes = getInput(path)
+    let originalEvents = (new IonEventStream(ion.makeReader(bytes))).getEvents();
+    let textWriter = ion.makeTextWriter();
+    textWriter.writeValues(ion.makeReader(bytes));
+    textWriter.close();
+    let textBytes = textWriter.getBytes();
+    let textEvents = (new IonEventStream(ion.makeReader(textBytes))).getEvents();
+    let binWriter = ion.makeBinaryWriter();
+    binWriter.writeValues(ion.makeReader(bytes));
+    binWriter.close();
+    let binBytes = binWriter.getBytes();
+    let binEvents = (new IonEventStream(ion.makeReader(binBytes))).getEvents();
+    for (let i = 0; i < originalEvents.length - 2; i += originalEvents[i].ionValue.length + 1) {
+        let event = originalEvents[i];
+        let textEvent = textEvents[i];
+        let binEvent = binEvents[i];
+        //if were equal everybody equals everybody
+        let comparisons : any = [];
```

Comment:
Comment please.  looks like each element in this array will be either:
- [original IonEventStream, text IonEventStream, binary IonEventStream]
- or [original ionValue, text ionValue, binary ionValue]

Or perhaps I'm confused as to what `event.ionValue[j]` actually contains...?

Reply:
Each top level value is a container (either sexp or list of strings). The ionValue is the backing array of inner IonEvents.

Reply:
We're missing a comment here that describes what the `comparisons` variable contains, as it is not obvious from reading the code, and it appears to differ between 'embedded_documents' and non-embedded_documents.

For an `IonEvent` representing a scalar value, `IonEvent.ionValue` appears to be the value returned by `Reader.value()`.  For an `IonEvent` representing a container, `IonEvent.ionValue` is an array of `IonEvents` representing the container start, contents, and end.

That said, each element of the `comparisons` array appears to be either:
* [IonEventStream (original), IonEventStream (text), IonEventStream (binary)]
* or [IonEvent.ionValue (original), IonEvent.ionValue (text), IonEvent.ionValue (binary)] -- where `IonEvent.ionValue` may be either a scalar value or an array of `IonEvents` (for containers)

Is that accurate?

Reply:
yes, both have the same comparison method equals().